app/controllers/webhooks_controller.rb

```diff
@@ -18,6 +18,16 @@ def state
     branches = params[:branches] || []
     if branches.find { |branch| branch[:name] == stack.branch }
       commit = stack.commits.find_by_sha!(params[:sha])
+      if (params[:state] == 'success')
+        if !commit.significant_status.state.success?
+          build_and_fire_status_webhook...
+        end
+      elsif (params[:state] != 'pending')
+        status = Status.new(params.permit(:state, :description, :target_url, :context, :created_at))
+        if commit.stack.filter_visible_statuses([status]) && commit.stack.filter_meaningful_statuses([status])
+          build_and_fire_status_webhook(status)
+        end
+      end
       commit.statuses.create!(params.permit(:state, :description, :target_url, :context, :created_at))
```

Comment:
Or you could refactor this `commit.statuses.create!` in `commit.record_status(*)`, and inside that method you can both check the `significant_status` before and after the new status is created.

If the meaningful status change it probably mean we should fire a hook.